### ISSUE TITLE
high cpu after make multiple files holding down <ctrl + v>

### ISSUE THREAD

**Issue by sc0w** [mate-desktop/caja#850]
#### Expected behaviour

caja must work more quickly, and no high cpu after the process

#### Actual behaviour

caja seems to work too slow, and after the process with high cpu

#### Steps to reproduce the behaviour

1. go to any foder

2. touch 1

3. <ctrl + c> over the file 1

5. keep hold down <ctrl + v> and try to make 1000 files

The process over the time is too slow, and when finish, caja seems to work always slow with high cpu

#### MATE general version

1.18.0

#### Package version

1.18.4-1

#### Linux Distribution

debian testing

**Comment by sc0w** [comment]
It doesn't happen with caja 1.16.6

**Comment by sc0w** [comment]
culprit PR: https://github.com/mate-desktop/caja/pull/568

**Comment by lukefromdc** [comment]
I could not quickly generate 1,000 copies but it looks like what is going on is that for a control-v cut and paste, but it looks like the attempt to rapidly make so many copies of 1 file (never seen this in normal use) overloads the file progress/queue dialog with so many initiations of it, completions of it, and restarts of it.

 Every interaction with the dailog is more code to run, and normally this is no problem but trying to execute 1,000 different file transactions at once could certainly create a high CPU load. If this code did not exist the same issue would exist, just requiring slower machines to be noticed as the codepath would be shorter. In a normal use where you either move a lot of files at once by DnD (and maybe run into another, unrelated issue), or interact with files sequentially at a rate no higher than one file every few seconds (with your brain and mouse hand saturated before anything else), this is simply not a problem.

**Comment by sc0w** [comment]
I suppose the high cpu while pressing <ctrl+v> is normal, but it is hard to make multiple files since https://github.com/mate-desktop/caja/pull/568

before the PR, it can be make it easily and quiclky, later it is too difficult

and after the process, caja works slowly with high cpu on every click

**Comment by monsta** [comment]
@emanuele-f: please have a look at this since it's related to #568.

**Comment by lukefromdc** [comment]
One possible workaround both for this and for the issue of one long file transfer requiring user confirmation of other, short jobs would be to use a gsettings preference to let users turn this on and off. Ideally this would also be accessable from the preferences gui, for users unfamiliar with dconf.

**Comment by monsta** [comment]
There's also a request for an option to disable the queue: https://github.com/mate-desktop/caja/issues/844

**Comment by emanuele-f** [comment]
Hi, I agree with @lukefromdc, maybe a solution to solve both this and #844 is to add a preference with a dropdown where the users can select a file size. Any file smaller than that size will be started immediately.

@sc0w I'm pc, which has an i3 cpu and SSD, keeping CTRL+v pressed shows caja process 50% CPU and the copy window keeps opening, updating at times, and closing. Anyway, after the copy, cpu goes normal and caja is responsive. How long does it take for you to have caja responsible again after the copy has been completed?

**Comment by sc0w** [comment]
I don't agree, fix this with a preference/gsettings key, is not a real fix

I suggest, for example: check the time between one copy and the next, and if it is less than 1 sec, disable the queuing/pausing

@emanuele-f 

> I'm pc, which has an i3 cpu and SSD, keeping CTRL+v pressed shows caja process 50% CPU and the copy window keeps opening, updating at times, and closing. Anyway, after the copy, cpu goes normal and caja is responsive.

you tried with 1000 files?

> How long does it take for you to have caja responsible again after the copy has been completed?

I waited 15 min and caja still works slowly with high cpu in every click (changing folders, using the scroll bar...), so I need to kill the process

**Comment by lukefromdc** [comment]
Has anyone checked this in Nemo to see if its an issue there too?

**Comment by sc0w** [comment]
@lukefromdc 

tests in nemo 3.4.7:

Gives me 1 min or less to create the 1000 files, and after, nemo works fine

**Comment by emanuele-f** [comment]
With 1000 files (seeeveral minutes) I experience the same caja slowness issues of @sc0w . 

Nevetheless, let's focus on the solution. I think measuring time is not a good solution because it does not solve the problem where you have a large file transfer in progress and still want to perform automatically the copy of small files. In this case, they will be queued anyway. I'd like to fix the two issues with a common solution.

**Comment by lukefromdc** [comment]
I think I found the difference in Nemo: the entire file progress dialog does not appear at all when doing this (for a few dozen copies at least, all I tested). I do not know how you are specifying 1,000 copies, I'm just holding down the ctrl-v key combination in a test directory.

**Comment by vkareh** [comment]
Fixed by #1896
